### PR TITLE
Add Non interactive fee bumping example

### PR DESCRIPTION
**Pull request by sanket1729**
Had to refactor some of the dummy_env stuff.

If I try to include change output in the default env, everything breaks and I did not want to spend time trying to fix all of tests. Instead, we create a new default with fee just for this one test case.

### CODE REVIEW

**Comment by apoelstra** [comment]
In 8231b9547fb1c064321df154fb9ed6b44a13468e: trailing whitespace on several lines.

**Comment by apoelstra** [comment]
In 8231b9547fb1c064321df154fb9ed6b44a13468e:

Looks like you're using `left_pad` in a couple places to widen integer types. You can just use `<u64>::from` or `<u32>::into`.

Your changes to the test environment seem fine to me.

**Comment by apoelstra** [comment]
Done reviewing 16ebe7411e78dd5e5d02b67d6ebc083976cd7dfd

**Comment by apoelstra** [comment]
cc @uncomputable can you provide some guidance here? It doesn't seem right that the user has to call a jet explicitly for integer promotion.

**Comment by sanket1729** [comment]
```
let v2 : u64 = <u1>::into(v1);
   |                    ^^^^^^^^^^^^^^ Cannot cast values of type `u1` as values of type `u64`
```

File: src/array.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ impl<'a, A> BTreeSlice<'a, A> {
     }
 }
 
-impl<'a, A: Clone> BTreeSlice<'a, A> {
+impl<A: Clone> BTreeSlice<'_, A> {
```
Comment (uncomputable) [inline src/array.rs:23]: ab42d67c78505463cf66b7f54a4137ea017e51d4: Are you using rust analyzer? `clippy` didn't show me these lints.

**Comment by uncomputable** [comment]
Casting is possible when two Simfony types are based on the same Simplicity type. Casting is a NOP in terms of Simplicity. Integers with different bit width are different types in Simplicity, so there is no cast.

To convert `u1` to `u64`, [`left_pad_low_1_64` can be called](https://docs.rs/simfony-as-rust/latest/simfony_as_rust/jet/multi_bit_logic/fn.left_pad_low_1_64.html). That seems to be the cheapest and most concise way. Alternatively, a nested tuple of integers and Booleans can be cast into `u64`. The cast itself is free, but constructing the tuple is not.

```
let b: bool = true;
let x: u64 = <(u32, (u16, (u8, (u4, (u2, (u1, bool))))))>::into((0, (0, (0, (0, (0, (0, b)))))));
assert!(jet::eq_64(x, 1));
```

**Review by uncomputable — APPROVED** [review]
ACK ab42d67c78505463cf66b7f54a4137ea017e51d4 Feel free to optimize the `check_neg`  function.

**Comment by uncomputable** [comment]
I have code to dynamically generate witness data for most of the Simfony examples, so we can change the Elements environment without every test failing. I will make a PR when the API is ready and integrate the hotfix from this PR.

**Review by apoelstra — APPROVED** [review]
ACK ab42d67c78505463cf66b7f54a4137ea017e51d4; successfully ran local tests

**Comment by sanket1729** [comment]
I am planning to post this example on delving bitcoin to get some more attention to simphony and some cool applications. Let me know if you have any objections

**Comment by uncomputable** [comment]
That's a great idea. Make sure you hit the right spelling :)

**Pull request merged**